alloc test (requires nightly) is broken

## STR

``` console
$ cd firmware/qemu
$ cargo +nightly run --bin alloc
error[E0658]: trait bounds other than `Sized` on const fn parameters are unstable
   --> /home/japaric/.cargo/registry/src/github.com-1ecc6299db9ec823/lock_api-0.3.4/src/mutex.rs:106:6
    |
106 | impl<R: RawMutex, T> Mutex<R, T> {
    |      ^
    |
    = note: see issue #57563 <https://github.com/rust-lang/rust/issues/57563> for more information
    = help: add `#![feature(const_fn_trait_bound)]` to the crate attributes to enable
```

## Meta

``` console
$ rustc +nightly -V
rustc 1.54.0-nightly (bacf770f2 2021-05-05)
```

--- a/firmware/qemu/Cargo.toml
+++ b/firmware/qemu/Cargo.toml
@@ -14,7 +14,7 @@ cortex-m = "0.7.2"
 cortex-m-rt = "0.6.13"
 cortex-m-semihosting = "0.3.7"
 alloc-cortex-m = { version = "0.4.1", optional = true }
-linked_list_allocator = { version = "0.8.11", optional = true }
+linked_list_allocator = { version = "0.9.0", optional = true }
 
 [features]
 defmt-default = [] # log at INFO, or TRACE, level and up

